File: tests/expectations/both/box.c
#ifdef __cplusplus
template <typename T>
using Box = T*;
#endif


#include <stdarg.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdlib.h>

typedef struct NotReprC_Box_i32 NotReprC_Box_i32;

typedef NotReprC_Box_i32 Foo;

typedef struct MyStruct {
  int32_t *number;
} MyStruct;

void drop_box(int32_t *x);

void root(const Foo *a, const MyStruct *with_box);

File: tests/expectations/box.zig
const std = @import("std");

pub const struct_NotReprC_____i32 = opaque {};
pub const NotReprC_____i32 = struct_NotReprC_____i32;
pub const Foo = NotReprC_____i32;
pub const MyStruct = extern struct {
    number: [*c]i32,
};
extern fn root(a: ?*const Foo, with_box: [*c]const MyStruct) anyopaque;
extern fn drop_box(x: [*c]i32) anyopaque;
extern fn drop_box_opt(x: [*c]i32) anyopaque;
